tests/utils-e2e.ts

```diff
@@ -1031,3 +1037,147 @@ export const constructE2ETests = (
     });
   });
 };
+
+export const testGasEstimation = async (
+  network: Network,
+  srcToken: Token,
+  destToken: Token,
+  amount: bigint,
+  swapSide: SwapSide,
+  dexKeys: string | string[],
+  contractMethod: ContractMethod,
+  targetDifference?: number,
+) => {
+  assert(
+    testingEndpoint,
+    'Estimation can only be tested with testing endpoint',
+  );
+  // initialize pricing
+  const sdk = new APIParaswapSDK(network, dexKeys);
+  await sdk.initializePricing();
+  // fetch the route
+  const priceRoute = await sdk.getPrices(
+    srcToken,
+    destToken,
+    amount,
+    swapSide,
+    contractMethod,
+  );
+  console.log({ priceRoute });
+  // prepare state overrides
+  const tenderlySimulator = TenderlySimulatorNew.getInstance();
+  // any address works
+  const userAddress = ethers.Wallet.createRandom().address;
+  // init `StateOverride` object
+  const stateOverride: StateOverride = {};
+  // add overrides for src token
+  if (srcToken.address.toLowerCase() === ETHER_ADDRESS) {
+    // add eth balance to user
+    stateOverride[userAddress] ||= {};
+    stateOverride[userAddress].balance = (amount * 2n).toString();
+  } else {
+    // add token balance and allowance to Augustus
+    // calculate the exact slots
+    const srcTokenSlots = await tenderlySimulator.getTokenStorageSlots(
+      network,
+      srcToken.address,
+    );
+    const userSrcBalanceOfSlot = tenderlySimulator.calculateAddressBalanceSlot(
+      srcTokenSlots.balanceSlot,
+      userAddress,
+      srcTokenSlots.isVyper,
+    );
+    const userSrcAllowanceToAugustusSlot =
+      tenderlySimulator.calculateAddressAllowanceSlot(
+        srcTokenSlots.allowanceSlot,
+        userAddress,
+        priceRoute.tokenTransferProxy,
+        srcTokenSlots.isVyper,
+      );
+    // add the overrides
+    stateOverride[srcToken.address] = {
+      storage: {
+        [userSrcBalanceOfSlot]: ethers.utils.defaultAbiCoder.encode(
+          ['uint'],
+          [amount * 2n],
+        ),
+        [userSrcAllowanceToAugustusSlot]: ethers.utils.defaultAbiCoder.encode(
+          ['uint'],
+          [amount * 2n],
+        ),
+      },
+    };
```

Comment:
I think we can move it to separate method in `TenderlySimulatorNew`

Reply:
added separate methods in https://github.com/paraswap/paraswap-dex-lib/pull/886/commits/7f7480e01233320cf4cf7c40fd405c2d355ecdf0